### ISSUE TITLE
Convergence issue! 

### ISSUE THREAD

**Issue by harshangrjn** [oxfordcontrol/Clarabel.jl#97]
Please find this SDP below. There seems to be an issue in convergence (using v0.2.0) with this error: 
```
LoadError: PosDefException: matrix is not positive definite; Cholesky factorization failed.
```
This can be circumvented by reducing the number of floating points in elements of A. Most probably a numerical issue. Here is the SDP instance: 
```
using JuMP 
using Clarabel

A = [0.0 11.08638213 4.23716998 3.42755487 15.4293342
     11.08638213 0.0 6.26171389 21.78641091 23.78527429 
     4.23716998 6.26171389 0.0 25.51211523 30.34857242 
     3.42755487 21.78641091 25.51211523 0.0 2.24851143 
     15.4293342 23.78527429 30.34857242 2.24851143 0.0]

n = size(A)[1]

m = JuMP.Model(Clarabel.Optimizer)
@variable(m, γ >= 0)
@variable(m, W[1:n, 1:n], PSD)
@variable(m, 0 <= x[1:n, 1:n] <= 1)
@objective(m, Max, γ)
@constraint(m, [i=1:(n-1), j=(i+1):n], W[i,j] == -A[i,j]*x[i,j] + γ/n)
@constraint(m, [i=1:n], W[i,i] == sum(A[i,:] .* x[i,:]) - γ*(n-1)/n)
@constraint(m, [i=1:n], sum(x[i,:]) >= 1)
@constraint(m, [i=1:(n-1), j=(i+1):n], x[i,j] == x[j,i])
@constraint(m, sum(x) == 2*(n-1))

JuMP.optimize!(m)
```

**Comment by goulart-paul** [comment]
Thanks.   We are aware of a number of similar issues that effect solver convergence at high tolerance.   We have implemented some improvements for v0.3 that solve the problem you provided, but haven't released them yet.

In the interim, you might have success if you relax the tolerances a bit, say `tol_gap_abs` and  `tol_gap_rel` at `1e-5` or so.

I suggest to keep this open until we make a public release that can solve the problem above.

**Comment by harshangrjn** [comment]
Thanks @goulart-paul. Didn't realize that I could adjust the tolerances. Also, you may be aware of another issue. If I add a few more linear constraints to the above SDP, a different error occurs as follows: 
```
LoadError: ArgumentError: matrix contains Infs or NaNs
```
Not sure if this is also due to the tolerances. Let me know if in case you need those constraints for testing purposes.

**Comment by goulart-paul** [comment]
It's possibly the same underlying problem.   The main issue is that v.0.2 is not doing enough to protect against poor conditioning at high tolerances.   So in the first case it ended up trying to factor a matrix with a very slightly negative eigenvalue.   In the second case it probably got a very ill conditioned KKT matrix to factor, and ended up with a division by zero.

That is only a guess though, and it would be nice to have an example problem for us to test against before the next release.

**Comment by harshangrjn** [comment]
Thanks @goulart-paul. I shall try and get back to you.

**Comment by goulart-paul** [comment]
This appears to be fixed by #114
